xmodule/modulestore/store_utilities.py

```diff
@@ -104,3 +106,23 @@ def get_draft_subtree_roots(draft_nodes):
     for draft_node in draft_nodes:
         if draft_node.parent_url not in urls:
             yield draft_node
+
+
+def derived_key(courselike_source_key, block_key, parent):
+    """
+    Return a new reproducible block ID for a given root, source block, and destination parent.
+
+    This is useful for ensuring that 'refreshable' blocks always have the same ID when updated.
```

Comment:
Would you mind adding a more verbose description of this use case to this docstring? This is the only occurrence of the "refreshable" term in the whole `edx-platform`.

Reply:
Done!